The script ../build_DORI_annotations_v1.py builds an annotation.csv file which contains both DORI and DCLDE data. The current version (v1) is a reduced set of the entire DORI set.
- DORI contains both pseudo labels assigned by a predictor as well as human generated labels. V1 only considers human generated labels.
- DORI contains clicks as well as pulsed calls - clicks are not included in V1


In [3]:
import pandas as pd
COMBINED = pd.read_csv("/home/noah/HALLO_encoder_collection/data_raw/DORI/20261005_183013/annotations.csv")
print("len of combined:", len(COMBINED))
DCLDE = pd.read_csv("/home/noah/HALLO_encoder_collection/data_raw/DCLDE_2027/20260921_131510/annotations.csv")
print("len of DCLDE:", len(DCLDE))
print("new rows:", len(COMBINED) - len(DCLDE))

/tmp/ipykernel_739423/4130181791.py:2: DtypeWarning: Columns (5,7,10,12,13,15,16,23,25,28,32,33,34,35,36,37,38,39,40) have mixed types. Specify dtype option on import or set low_memory=False.
  COMBINED = pd.read_csv("/home/noah/HALLO_encoder_collection/data_raw/DORI/20261005_183013/annotations.csv")


len of combined: 1839561
len of DCLDE: 1180187
new rows: 659374


/tmp/ipykernel_739423/4130181791.py:4: DtypeWarning: Columns (5,7,10,12,13,15,16,25) have mixed types. Specify dtype option on import or set low_memory=False.
  DCLDE = pd.read_csv("/home/noah/HALLO_encoder_collection/data_raw/DCLDE_2027/20260921_131510/annotations.csv")


MOST annotationss are removed when taking away the pseudo labels, before removal:
species_label_clean
orca                            15050
humpback                         2793
pacific white sided dolphin      2579
sea lion                         1001
false killer whale                244
multiple classes                  216
noise                             109
sperm whale                        66
northern right whale dolphin        4
minke                               3
gray                                1

After:
species_label_clean
orca        7610
humpback       1


In [ ]:
COMBINED['Labels'].value_counts() - DCLDE['Labels'].value_counts()



Labels
Background    652228
HW                 0
SRKW            5174
TKW             1830
UndBio             0
AB                 1
NRKW              14
SAR                0
KW_und             0
OKW              127
Name: count, dtype: int64

We see:
652228 Background samples
5174 SRKW
1830 TKW
14 NRKW
127 OKW


Tekteksen - indigenous name for East Point

In [ ]:
# COMBINED[COMBINED['Dataset'] == 'DORI VENUS (Strait of Georgia)']['Labels'].value_counts()
# COMBINED[COMBINED['Provider'] == 'SIMRES']['Dataset'].value_counts()
for ds in ['DORI VENUS (Strait of Georgia)', 'DORI VENUS (Saanich)', 'LimeKiln', 'Tekteksen']:
    print(ds)
    print(COMBINED[COMBINED['Dataset'] == ds]['Labels'].value_counts())

DORI VENUS (Strait of Georgia)
Labels
Background    490756
SRKW            3674
TKW             1176
AB                 1
Name: count, dtype: int64
DORI VENUS (Saanich)
Labels
Background    416
TKW             4
Name: count, dtype: int64
LimeKiln
Labels
SRKW          760
KW_und        576
Background    519
HW            140
AB             97
UndBio         94
Name: count, dtype: int64
Tekteksen
Labels
Background    6955
SRKW          3418
AB             419
KW_und         160
HW              21
Name: count, dtype: int64


In [20]:
datasets = ['DORI VENUS (Strait of Georgia)', 'DORI VENUS (Saanich)', 'LimeKiln', 'Tekteksen']
subset = COMBINED[COMBINED['Dataset'].isin(datasets)]
counts = pd.crosstab(subset['Labels'], subset['Dataset'], margins=True, margins_name='Total')
counts = counts[datasets + ['Total']]
counts = pd.concat([counts.drop('Total').sort_values('Total', ascending=False), counts.loc[['Total']]])
counts

Dataset,DORI VENUS (Strait of Georgia),DORI VENUS (Saanich),LimeKiln,Tekteksen,Total
Labels,,,,,
Background,490756,416,519,6955,498646
SRKW,3674,0,760,3418,7852
TKW,1176,4,0,0,1180
KW_und,0,0,576,160,736
AB,1,0,97,419,517
HW,0,0,140,21,161
UndBio,0,0,94,0,94
Total,495607,420,2186,10973,509186
